### PR TITLE
Dynamic Catalog Integration with JSON Data

### LINKED ISSUE #2: Make the Catalog Page Dynamic Using Recipe JSON
**Description**:
Refactor the `catalog.html` page to dynamically render catalog from the recipe JSON file while maintaining the current design and styling.

### Tasks  
1. **Integrate with Catalog Page**  
   - Replace hardcoded content in `catalog.html` with dynamic content generated from the JSON file.  
   - Use JavaScript to dynamically create and populate catalog items based on JSON data.
2. **Styling Consistency**  
   - Preserve the existing styling defined in `catalog.css`.  
   - Test responsiveness and adjust CSS as needed to ensure the page displays correctly across devices.

#### Deliverables  
1. Updated `catalog.html` with placeholders for dynamic rendering.  
2. A JavaScript script to fetch, parse, and render recipe data dynamically.  
3. A fully functional and styled catalog page with dynamic content from the recipe JSON.

Ensure the code is modular, maintainable, and ready for future enhancements.

### PR DESCRIPTION
**Pull request by Keerthivasan-Venkitajalam**
Replaced hardcoded content in `catalog.html` with dynamic recipe rendering from `recipes.json`.  
- **Key Updates**:  
  - JavaScript dynamically fetches and generates recipe cards with details like name, time, and ratings.  
  - Maintained styling and ensured responsiveness across devices.  
  - Modularized JavaScript for easy scalability and future features.  

Tested for proper layout, styling consistency, and smooth performance on various screen sizes.

### CODE REVIEW

File: JS files/catalog.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                        <h2></h2>
+                        <div class="tags_div">
+                            <div class="tag">${recipe.time}</div>
+                        </div>
+                        <div class="review" style="color: rgb(243, 181, 25)">
+                            ${generateStars(recipe.reviews.user_reviews)}
+                        </div>
+                    </div>
+                    <a href="recipe.html?id=${recipe.id}">
+                        <i class="fas fa-shopping-basket basket"></i>
```
Comment (amri-tah) [inline JS files/catalog.js]: Please just change this icon to a heart. There's too much gap between the cards so if you could do something to fix that it'd be great too. Nice work @Keerthivasan-Venkitajalam :)

Comment (Keerthivasan-Venkitajalam) [reply JS files/catalog.js]: Will do that :) 👍🏻

**Comment by Keerthivasan-Venkitajalam** [comment]
Changes made according to Feedback @amri-tah

**Review by amri-tah — APPROVED** [review]
LGTM! Really love the attention to detail :) @Keerthivasan-Venkitajalam

**Comment by Keerthivasan-Venkitajalam** [comment]
> LGTM! Really love the attention to detail :) @Keerthivasan-Venkitajalam

Thank you @amri-tah :)

**Pull request merged**